src/templates/_image.tpl

```diff
@@ -0,0 +1,22 @@
+{{/*
+Builds a full OCI image reference from the given image object.
+
+Usage:
+  {{ include "papermc-server.image" (dict "image" .Values.container.image) }}
+
+If digest is "none", returns: <registry>/<name>:<tag>
+Otherwise returns: <registry>/<name>@<digest>
+*/}}
+{{ define "papermc-server.image" -}}
+{{ $image := .Values.container.image -}}
+{{ $registry := $image.registry | default "docker.io" -}}
+{{ $name := $image.name | default "djaytan/papermc-server" -}}
+{{ $tag := $image.tag | default "latest" -}}
+{{ $digest := $image.digest | default "none" -}}
```

Comment:
That's weird to have as default `none`.

Having an undefined variable is fine and a good way to state a variable as "undefined".